src/EnergyPlus/UnitarySystem.cc

```diff
@@ -13268,22 +13268,24 @@ namespace UnitarySystems {
                                                                                                       singleMode);
                             if ((state.dataLoopNodes->Node(OutletNode).HumRat - DesOutHumRat) < Acc) break;
                         }
+                        // make sure outlet HumRat is below set point before calling SolveRoot
+                        if (state.dataLoopNodes->Node(OutletNode).HumRat < DesOutHumRat) {
+                            std::array<Real64, 8> Par2 = {
+                                double(this->m_CoolingCoilIndex),
+                                DesOutHumRat,
+                                // dehumidification mode = 0 for normal mode, 1+ for enhanced mode
+                                // need to test what happens when Alt mode doesn't exist, or somehow test for it,
+                                // or fatal out in GetInput
+                                0.0, // DehumidMode
+                                double(FanOpMode),
+                                double(this->m_CoolingSpeedNum),
+                                1.0, //  this->m_CoolingSpeedRatio;
+                                1.0, // run on latent, check coil outlet node HumRat
+                                0.0  // dummy because genericDXCoilResidual takes 8 parameters
+                            };
 
-                        std::array<Real64, 8> Par2 = {
-                            double(this->m_CoolingCoilIndex),
-                            DesOutHumRat,
-                            // dehumidification mode = 0 for normal mode, 1+ for enhanced mode
-                            // need to test what happens when Alt mode doesn't exist, or somehow test for it,
-                            // or fatal out in GetInput
-                            0.0, // DehumidMode
-                            double(FanOpMode),
-                            double(this->m_CoolingSpeedNum),
-                            1.0, //  this->m_CoolingSpeedRatio;
-                            1.0, // run on latent, check coil outlet node HumRat
-                            0.0  // dummy because genericDXCoilResidual takes 8 parameters
-                        };
-
-                        General::SolveRoot(state, HumRatAcc, MaxIte, SolFla, PartLoadFrac, &this->genericDXCoilResidual, 0.0, 1.0, Par2);
+                            General::SolveRoot(state, HumRatAcc, MaxIte, SolFla, PartLoadFrac, &this->genericDXCoilResidual, 0.0, 1.0, Par2);
+                        }
```

Comment:
SolveRoot was failing with SolFla = -2 when coil could not get to set point. Only call SolveRoot when a solution can be found.